python/ray/serve/utils.py

```diff
@@ -365,3 +366,68 @@ def get_current_node_resource_key() -> str:
                     return key
     else:
         raise ValueError("Cannot found the node dictionary for current node.")
+
+
+def register_custom_serializers():
+    """Install custom serializers needed for Ray Serve."""
+    import starlette.datastructures
+    import pydantic.fields
+
+    assert ray.is_initialized(
+    ), "This functional must be ran with Ray initialized."
+
+    # Pydantic's Cython validators are not serializable.
+    # https://github.com/cloudpipe/cloudpickle/issues/408
+    ray.worker.global_worker.run_function_on_all_workers(
+        lambda _: ray.util.register_serializer(
+            pydantic.fields.ModelField,
+            serializer=lambda o: {
+                "name": o.name,
+                "type_": o.type_,
+                "class_validators": o.class_validators,
+                "model_config": o.model_config,
+                "default": o.default,
+                "default_factory": o.default_factory,
+                "required": o.required,
+                "alias": o.alias,
+                "field_info": o.field_info,
+            },
+            deserializer=lambda kwargs: pydantic.fields.ModelField(**kwargs),
```

Comment:
@simon-mo can we upstream this into Ray so we don't need to dynamically register this callback?